File: src/main/java/com/qtech/bubbles/common/ability/AbilityTriggerType.java
package com.qtech.bubbles.common.ability;

import com.qtech.bubbles.common.RegistryEntry;
import com.qtech.bubbles.common.init.ObjectInit;

import java.util.Objects;

public class AbilityTriggerType extends RegistryEntry implements ObjectInit<AbilityTriggerType> {
    public static final AbilityTriggerType KEY_TRIGGER = new AbilityTriggerType();
    private final long hash;

    public AbilityTriggerType() {
        this.hash = System.currentTimeMillis();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        AbilityTriggerType that = (AbilityTriggerType) o;
        return hash == that.hash;
    }

    @Override
    public int hashCode() {
        return Objects.hash(super.hashCode(), hash);
    }
}

File: src/main/kotlin/qtech/bubbles/common/ability/AbilityTriggerType.kt
package qtech.bubbles.common.ability

import qtech.bubbles.common.RegistryEntry
import qtech.bubbles.common.init.ObjectInit
import java.util.*

class AbilityTriggerType : RegistryEntry(), ObjectInit<AbilityTriggerType?> {
    private val hash: Long = System.currentTimeMillis()
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        if (!super.equals(other)) return false
        val that = other as AbilityTriggerType
        return hash == that.hash
    }

    override fun hashCode(): Int {
        return Objects.hash(super.hashCode(), hash)
    }

    companion object {
        var KEY_TRIGGER = AbilityTriggerType()
    }
}